tests/Unit/Domain/Test_BlockLogicalCoordinates.cpp

```diff
@@ -0,0 +1,523 @@
+  auto element_ids_1 = initial_element_ids<3>(1, {{2, 1, 1}});
+  auto element_ids_2 = initial_element_ids<3>(2, {{0, 2, 1}});
+  auto element_ids_3 = initial_element_ids<3>(3, {{2, 2, 2}});
+  auto element_ids_6 = initial_element_ids<3>(6, {{2, 2, 2}});
+  auto element_ids_5 = initial_element_ids<3>(5, {{1, 2, 0}});
+  auto element_ids_7 = initial_element_ids<3>(7, {{2, 0, 1}});
+  std::copy(element_ids_0.begin(), element_ids_0.end(),
+            std::back_inserter(element_ids));
+  std::copy(element_ids_1.begin(), element_ids_1.end(),
+            std::back_inserter(element_ids));
+  std::copy(element_ids_2.begin(), element_ids_2.end(),
+            std::back_inserter(element_ids));
+  std::copy(element_ids_3.begin(), element_ids_3.end(),
+            std::back_inserter(element_ids));
+  std::copy(element_ids_6.begin(), element_ids_6.end(),
+            std::back_inserter(element_ids));
+  std::copy(element_ids_5.begin(), element_ids_5.end(),
+            std::back_inserter(element_ids));
+  std::copy(element_ids_7.begin(), element_ids_7.end(),
+            std::back_inserter(element_ids));
+
+  // I (Mark) computed these expected quantities by hand, given the
+  // points above and the choices of elements.
+  std::vector<size_t> expected_id_indices{
+      {1, 8, 29, 84, 92, 153, 169, 225, 239, 215}};
+  // The last point above is on an element boundary;
+  // element_logical_coordinates should choose the first element in
+  // the list of elements it is passed.
+  std::vector<std::vector<size_t>> expected_offset{
+      std::vector<size_t>{6}, std::vector<size_t>{0, 1}, std::vector<size_t>{2},
+      std::vector<size_t>{3}, std::vector<size_t>{4},    std::vector<size_t>{5},
+      std::vector<size_t>{7}, std::vector<size_t>{8},    std::vector<size_t>{9},
+      std::vector<size_t>{10}};
+  std::vector<std::vector<std::array<double, 3>>> expected_elem_log{
+      std::vector<std::array<double, 3>>{{{-0.6, -0.2, 0.2}}},
+      std::vector<std::array<double, 3>>{{{0.6, 0.6, 0.6}},
+                                         {{-0.2, -0.2, -0.2}}},
+      std::vector<std::array<double, 3>>{{{0.84, 0.84, 0.84}}},
+      std::vector<std::array<double, 3>>{{{-0.6, 0.92, 0.92}}},
+      std::vector<std::array<double, 3>>{{{-0.04, -0.2, 0.92}}},
+      std::vector<std::array<double, 3>>{{{-0.6, -0.2, 0.84}}},
+      std::vector<std::array<double, 3>>{{{0.6, -0.2, 0.2}}},
+      std::vector<std::array<double, 3>>{{{0.6, 0.2, -0.2}}},
+      std::vector<std::array<double, 3>>{{{-0.6, 0.6, 0.2}}},
+      std::vector<std::array<double, 3>>{{{1.0, 1.0, 1.0}}}};
+
+  test_block_logical_coordinates(
+      domain, x_frame, expected_block_ids, expected_x_logical, element_ids,
+      expected_id_indices, expected_offset, expected_elem_log);
+}
+
+template <size_t Dim>
+void test_block_logical_coord_holder() {}
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.BlockLogicalCoords", "[Domain][Unit]") {
```

Comment:
If I understood the test correctly it also tests the ElementLogicalCoords. If that is the case then maybe name it `BlockAndElementLogicalCoords`?